backend/kemomimi_api/src/routes/public_items/index.rs

```diff
@@ -0,0 +1,200 @@
+use crate::libs::ApiImpl;
+use axum::{async_trait, extract::Host, http::Method};
+use axum_extra::extract::CookieJar;
+use chrono::Utc;
+use openapi::{
+    apis::public_items::{
+        PublicItems, PublicItemsGetResponse, PublicItemsPostResponse,
+        PublicItemsPublicItemIdDeleteResponse, PublicItemsPublicItemIdGetResponse,
+        PublicItemsPublicItemIdPutResponse,
+    },
+    models::{self, PublicItem},
+};
+use sqlx::{query_as, types::Uuid};
+use tracing::info;
+
+#[derive(Debug, Clone, PartialEq)]
+struct PublicItemRaw {
+    /// 備品のユニークID
+    pub public_item_id: Uuid,
+
+    /// 備品名
+    pub name: String,
+    pub category_id: Option<String>,
+
+    /// 備品の購入コスト
+    pub cost: Option<i32>,
+
+    /// 承認日
+    pub approval_date: Option<sqlx::types::time::Date>,
+
+    /// 耐用期限
+    pub expiration_date: Option<sqlx::types::time::Date>,
+
+    /// 現存しているか
+    pub is_remaining: bool,
+
+    // pub main_user_id: Option<String>,
+    /// 備考欄
+    pub remarks: Option<String>,
+}
+
+#[async_trait]
+impl PublicItems for ApiImpl {
+    #[doc = " 備品一覧取得."]
+    #[doc = ""]
+    #[doc = " PublicItemsGet - GET /public-items"]
+    #[must_use]
+    #[allow(
+        elided_named_lifetimes,
+        clippy::type_complexity,
+        clippy::type_repetition_in_bounds
+    )]
+    #[tracing::instrument]
+    async fn public_items_get<'life0>(
+        &'life0 self,
+        _method: Method,
+        _host: Host,
+        _cookies: CookieJar,
+        query_params: models::PublicItemsGetQueryParams,
+    ) -> Result<PublicItemsGetResponse, ()>
+    where
+        'life0: 'async_trait,
+    {
+        let data = query_as!(
+            PublicItemRaw,
+            r#"
+            SELECT
+                pi.public_item_id as "public_item_id!",
+                pi.name as "name!",
+                c.category_id as "category_id?",
+                pi.cost as "cost?",
+                pi.purchase_date as "approval_date?",
+                pi.expiration_date as "expiration_date?",
+                pi.is_remaining as "is_remaining!",
+                pi.remarks as "remarks?"
+            FROM
+                public_item pi
+            LEFT JOIN
+                product p ON pi.product_id = p.product_id
+            LEFT JOIN
+                 product_category pc ON p.product_id = pc.product_id
+            LEFT JOIN
+                category c ON pc.category_id = c.category_id
```

Comment:
LEFT JOIN は FROM 句の一部なのでもう1段字下げしたい

postgres なら using 使えるので使っていきましょう

```suggestion
            FROM
                public_item pi
                LEFT JOIN product p USING (product_id)
                LEFT JOIN product_category pc USING (product_id)
                LEFT JOIN category c USING (category_id)
```